File: back-end-application/game-service/src/main/java/be/technifutur/game/metier/service/game/GameService.java
package be.technifutur.game.metier.service.game;

import be.technifutur.game.models.dto.DetailedGameDTO;
import be.technifutur.game.models.dto.GameDTO;
import be.technifutur.game.models.entities.Genre;
import be.technifutur.game.models.forms.GameInsertForm;
import be.technifutur.game.models.forms.GameUpdateForm;

import java.util.List;
import java.util.UUID;

public interface GameService {

    List<DetailedGameDTO> getGames();

    DetailedGameDTO getGameByReference(UUID reference);

    DetailedGameDTO getGameByTitle(String title);

    GameDTO insertGame(GameInsertForm gameForm);

    GameDTO updateGame(UUID reference, GameUpdateForm gameForm);

    DetailedGameDTO deleteGame(UUID reference);

    GameDTO updateDeveloperOfGame(UUID gameReference, UUID devReference);

    GameDTO updateEditorOfGame(UUID gameReference, UUID edReference);

    void insertGameList(List<GameInsertForm> gameList);

    List<DetailedGameDTO> getGamesByGenre(Genre genre);
}

File: front-end-application/tp4-mist-angular/src/app/_services/game.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Game } from '../_models/game.model';
import { GameInsertForm } from '../_models/gameInsertForm.model';
import { GameUpdateForm } from '../_models/gameUpdateForm.model';

@Injectable({
  providedIn: 'root'
})
export class GameService {

  private readonly BASE_URL = "http://10.27.1.17:8080/game";

  constructor(private client: HttpClient) { }


  // --- GET ---
  getGames(){
    return this.client.get<Game[]>(this.BASE_URL);
  }

  getGameByReference(reference: String){
    return this.client.get<Game>(this.BASE_URL, {params: {reference: <string>reference} });
  }

  getGameByTitle(title: String){
    return this.client.get<Game>(this.BASE_URL, {params: {title: <string>title} });
  }

  // --- POST ---
  insertGame(form: GameInsertForm){
    return this.client.post<Game>(this.BASE_URL + "/add", form);
  }

  // --- PUT ---
  updateGame(form: GameUpdateForm){
    return this.client.put<Game>(this.BASE_URL + "/update", form);
  }

  // --- DELETE ---
  deleteGame(reference: String){
    return this.client.delete<Game>(this.BASE_URL + "/delete", {params: {reference: <string>reference} });
  }

  // --- UPDATE DEVELOPER & EDITOR ---
  updateDeveloperOfGame(reference: String, devReference: String){
    return this.client.patch<Game>(this.BASE_URL + "/updateDeveloper", {params: {reference: <string>reference} } , {params: {devReference: <string>devReference} });
  }

  updateEditorOfGame(reference: String, editReference: String){
    return this.client.patch<Game>(this.BASE_URL + "/updateEditor", {params: {reference: <string>reference}}, {params: {editReference: <string>editReference}});
  }

}
